File: codeforces/337A/cpp/337a.cpp
#include <iostream>
#include <algorithm>
using namespace std;

const int c_max = 50;

int main()
{
    int f[c_max] = {0};
    int n, m;
    cin >> n >> m;
    for (int i = 0; i < m; i++) {
        cin >> f[i];
    }
    sort(f, f+m);

    int result = 1000;
    for (int i = 0; i <= m-n; i++) {
        result = min(result, f[i+n-1] - f[i]);
    }
    cout << result << endl;
    return 0;
}

File: codeforces/337A/py/337a.py
import sys

n, m = map(int, sys.stdin.readline().split())
f = list(map(int, sys.stdin.readline().split()))
f.sort()

result = sys.maxsize
for i in range(0, m-n+1):
    result = min(result, f[i+n-1] - f[i])

print(result)
